internal/gengapic/testdata/empty_client_init.want

```diff
@@ -100,14 +100,38 @@ func (c *gRPCClient) Close() error {
 // Methods, except Close, may be called concurrently. However, fields must not be modified concurrently with method calls.
 type restClient struct {
 	host string
+	httpClient http.Client
 }
 
 func (c *restClient) Zip(ctx context.Context, req *mypackagepb.Bar, opts ...gax.CallOption) (*mypackagepb.Foo, error) {
-	m := jsonpb.Marshaler{}
-	if jsonReq, err := m.MarshalToString(req); err != nil {
+	// The default (false) for the other options are fine.
+	marshaler := protojson.MarshalOptions{AllowPartial: true, Multiline: true, EmitUnpopulated: true}
+	if jsonReq, err := marshaler.Marshal(req); err != nil {
 		return nil, err
+	}
+
+	url := fmt.Sprintf("https://%s/zip", c.host)
+	httpReq, err := http.NewRequestWithContext(ctx, "GET", url, bytes.NewReader(jsonReq))
+	if err != nil {
+		return nil, err
+	}
 
+	httpRsp, err := client.Do(httpReq)
+	defer httpRsp.Body.Close()
+	if err != nil{
+		return nil, err
+	} else if httpRsp.StatusCode >= 400 {
+		return nil, errors.E(nil, httpRsp.Status)
 	}
-	return nil, nil
 
+	buf := new(bytes.Buffer)
```

Comment:
If we need to read everything into memory maybe just https://pkg.go.dev/io/ioutil#ReadAll?